File: chaeeun/Programmers/Lv2/다음 큰 숫자.js
function solution(n) {
    var answer = 0;
    
    let oneCount = n.toString(2).match(/1/g).length
    
    while (true) {
        n++
        if (oneCount == n.toString(2).match(/1/g).length) {
            return n
        }
    }
    
    // return answer;
}

File: hyesu/SwiftAlgorithm/SwiftAlgorithm/Lv2/다음 큰 숫자.swift
//
//  다음 큰 숫자.swift
//  SwiftAlgorithm
//
//  Created by 김혜수 on 2023/02/18.
//

func solution(_ n: Int) -> Int {
    
    var answer: Int = n
    
    while true {
        let nCount = String(n, radix: 2).filter { $0 == "1" }.count
        answer += 1
        if nCount == String(answer, radix: 2).filter({ $0 == "1" }).count {
            return answer
        }
    }
    
    return answer
}
